composables/useToken.ts

```diff
@@ -1,74 +1,73 @@
 import { useFiatStore } from '@/stores/fiat'
-import { type ChainProperties, type Prefix } from '@kodadot1/static'
-import { chainPropListOf } from '@/utils/config/chain.config'
-import { groupByNestedProperty } from '@/utils/objects'
-import { availablePrefixes } from '@/utils/chain'
+import { CHAINS, type Prefix } from '@kodadot1/static'
+import { useIdentityStore } from '@/stores/identity'
+import { networkToPrefix } from '@/utils/chain'
 
 export interface TokenDetails {
   symbol: string
   value: number | string | null
   icon: string
   chains: Prefix[]
+  defaultChain: Prefix
+  tokenDecimals: number
 }
 
+const getAssetToken = (asset) => asset?.token || 'KSM'
+const getUniqueArrayItems = (items: string[]) => [...new Set(items)]
+
 export default function useToken() {
   const { getCurrentTokenValue } = useFiatStore()
   const { getTokenIconBySymbol } = useIcon()
+  const { getAvailableAssets } = useIdentityStore()
 
-  const availableChains = availablePrefixes().map(
-    (item) => item.value as Prefix
+  const availableAssets = computed(() => getAvailableAssets)
+  const availableTokensAcrossAllChains = computed(() =>
+    getUniqueArrayItems(Object.values(availableAssets.value).map(getAssetToken))
   )
 
-  const availableTokens = ['BSX', 'DOT', 'KSM']
-
-  const chainsProperties = computed(() => {
-    return availableChains.reduce(
-      (reducer, chain: Prefix) => ({
-        ...reducer,
-        [chain]: chainPropListOf(chain),
-      }),
-      {}
-    ) as { [k in Prefix]: ChainProperties }[]
-  })
-
-  const groupedTokensByChains = computed(() =>
-    groupByNestedProperty(chainsProperties.value, 'tokenSymbol')
-  )
+  const getTokenChains = (token: string): Prefix[] => {
+    const chains = Object.values(availableAssets.value).reduce(
+      (reducer: Prefix[], asset) => {
+        const assetToken = getAssetToken(asset)
+        const chainName = asset.chain
+        if (token === assetToken) {
+          return [...reducer, networkToPrefix[chainName] as Prefix]
+        }
+        return reducer
+      },
+      []
+    )
 
-  const getTokenChain = (token: string): Prefix[] =>
-    groupedTokensByChains.value[token] || []
+    return chains
+  }
 
   const tokens = computed<TokenDetails[]>(() => {
-    const filteredTokens = Object.keys(groupedTokensByChains.value).filter(
-      (token) => availableTokens.includes(token)
-    )
+    return availableTokensAcrossAllChains.value.map((tokenSymbol) => {
+      const chains = getTokenChains(tokenSymbol)
+      const defaultChain = chains[0]
 
-    return filteredTokens.map((tokenSymbol) => {
       return {
         symbol: tokenSymbol as string,
         value: getCurrentTokenValue(tokenSymbol),
         icon: getTokenIconBySymbol(tokenSymbol),
-        chains: getTokenChain(tokenSymbol),
+        chains: chains,
+        defaultChain: defaultChain,
+        tokenDecimals: CHAINS[defaultChain].tokenDecimals,
```

Comment:
It makes no sense that provides a 'random' value from the list 
`chains[0]`

We already have the default decimals for the current network.
```
const { decimals } = useChain()
```

Reply:
from what I understand this wouldn't work if you wanted to get the `tokenDecimals` value in the case of KSM on BSX

<img width="644" alt="CleanShot 2023-08-23 at 09 54 23@2x" src="https://github.com/kodadot/nft-gallery/assets/44554284/591e6541-c8d9-4395-8666-b118e796f777">

Reply:
> from what I understand this wouldn't work if you wanted to get the tokenDecimals value in the case of KSM on BSX

I mean, we should not randomly choose one chain to be the default chain of a token.  It should be a config that we manually set. 


> tokens across chains have the same tokenDecimals right ?
It's wrong to my knowledge.

Reply:
ok let me change this real quick

Reply:
fixed